src/Persistence/Search/SearchIndexFieldsBuilder.php

```diff
@@ -25,20 +25,25 @@ public function __construct(
 	 * @return array<string, SearchIndexField>
 	 */
 	public function createFields(): array {
-		$fields = [];
-
-		foreach ( $this->config->getFacets()->asArray() as $facetConfig ) {
-			$fieldType = $this->getFieldTypeForPropertyId( $facetConfig->propertyId );
+		try {
+			$itemType = $this->config->getInstanceOfId();
+		} catch ( Exception ) {
+			return [];
+		}
```

Comment:
If InstanceOfId is missing, we throw an exception. However, this will cause indexing to break, for example if the user enabled this extension but did not start using it yet.
On the other hand, if InstanceOfId is missing, then there is probably no point in indexing anything else, because our tabs will not be able to filter by type.

Reply:
Checking for the exception is weird.

`$this->config->hasInstanceOfId()` is better

But why bind to that detail. Could do `$this->config->isComplete()` or similar

Going further, maybe this service should not know about config potentially incomplete enough to prevent the extension from functioning. Maybe the code that executes this service could only be registered if the config is complete? This could also apply to other parts of the extension.